scvi/dataset/dataset.py

```diff
@@ -1090,6 +1117,21 @@ def map_cell_types(
         for cell_types, new_cell_type_name in cell_types_dict.items():
             self.merge_cell_types(cell_types, new_cell_type_name)
 
+    def reorder_cell_types(self, new_order: Union[List[str], np.ndarray]):
+        """Reorder in place the cell-types. The cell-types provided will be added at the beginning of `cell_types`
+        attribute, such that if some existing cell-types are omitted in `new_order`, they will be left after the
+        new given order
+
+        :param new_order:
+        """
+        for cell_type in self.cell_types:
+            if cell_type not in new_order:
+                new_order.append(cell_type)
+
+        cell_types = OrderedDict([((x,), x) for x in new_order])
+        self.map_cell_types(cell_types)
+        self.remap_categorical_attributes(["labels"])
```

Comment:
Is this call to `remap_categorical_attributes` really necessary? Isn't the remapping done in `map_cell_types` ?

Reply:
No it's not done and I think it's better like that. If the user starts using `map_cell_types` it mean the user knows what he's doing with the cell types and we shouldn't change labels under its nose.

Reply:
Agreed